src/reconnectcontroller/ReconnectController.ts

```diff
@@ -16,7 +18,7 @@ export default interface ReconnectController {
    * can be set.
    * @param{boolean} isFirstConnection whether this is the first attempt to connect for this session
    */
-  startedConnectionAttempt(isFirstConnection: boolean): void;
+  startedConnectionAttempt(isFirstConnection: boolean, idelMonitor: IdleMonitor): void;
```

Comment:
I don't think it's a good idea to depend on `IdleMonitor` in the interface. If a builder implements `ReconnectController` or already has implemented it, they need to pass an `IdleMonitor` instance to `startedConnectionAttempt()`. From the API design perspective, what does `IdleMonitor` mean to `startedConnectionAttempt()`?

I think you can implement this feature without modifying `ReconnectController` and `DefaultReconnectController`. Instead, you can just use the reconnect controller to turn on and off from the outside. (I remember we discussed this before.)

I will comment in `DefaultReconnectController`.